components/script/dom/element.rs

```diff
@@ -505,8 +498,19 @@ impl Element {
             self.node.owner_doc().register_shadow_root(&*shadow_root);
         }
 
+        self.upcast::<Node>().dirty(NodeDamage::OtherNodeDamage);
+
         Ok(shadow_root)
     }
+
+    pub fn detach_shadow(&self) {
+        if let Some(ref shadow_root) = self.shadow_root() {
+            self.node.owner_doc().unregister_shadow_root(shadow_root);
+            self.ensure_rare_data().shadow_root = None;
```

Comment:
This is not enough, this should also remove the containing shadow pointers and such, and clear the shadow root host pointer, clear the connected flags in shadow root and children, etc...